### PR TITLE
Adjust the sql query for total surplus

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Fixes https://github.com/cowprotocol/services/issues/3089

# Changes

Includes orders from "jit_orders" to calculate the total surplus.

- [ ] Updated sql query.

## How to test
Executed manually the new query against prod db. Two things are verified:
1. https://explorer.cow.fi/address/0xbf8868b754a77e90ea68ffc0b5b10a7c729457e1 now returns non-zero surplus.
2. Regular owners surplus remains the same, so this change did not break existing functionality.

Performance should be good since "jit_orders" table also has index on owners.

### CODE REVIEW

File: crates/orderbook/src/database/total_surplus.rs [hunk 1]
```diff
@@ -15,6 +15,21 @@ WITH regular_orders AS (
...
+    -- use this weird construction instead of `where owner=address or sender=address` to help postgres make efficient use of indices
+    WHERE uid = ANY(ARRAY_CAT((SELECT ids FROM regular_orders), (SELECT ids FROM onchain_orders)))
+    
+    UNION ALL
+    
+    -- Get orders from the jit_orders table that match the owner criteria but are not in orders
+    SELECT j.uid, j.kind, j.buy_amount, j.sell_amount, j.fee_amount, j.buy_token, j.sell_token
+    FROM jit_orders j
+    LEFT JOIN orders o ON j.uid = o.uid
+    WHERE j.owner = $1 AND o.uid IS NULL
```
Comment (squadgazzz) [inline crates/orderbook/src/database/total_surplus.rs:31]: Joining the tables seems to be redundant here.
```suggestion
    WHERE j.owner = $1 
      AND NOT EXISTS (
            SELECT 1 
            FROM orders o 
            WHERE o.uid = j.uid
        )
```

**Review by MartinquaXD — COMMENTED** [review]
Change looks alright but agree with @squadgazzz's comment.
> Performance should be good since "jit_orders" table also has index on owners.

Can you please compare the performance of the surplus query for an account that has many regular orders by manually executing the old and the new query?
Would be a shame if we pessimise the performance of the common use case by adding support for jit orders.

File: crates/orderbook/src/database/total_surplus.rs [hunk 2]
```diff
@@ -15,6 +15,21 @@ WITH regular_orders AS (
...
+    SELECT uid, kind, buy_amount, sell_amount, fee_amount, buy_token, sell_token
+    FROM orders
+    -- use this weird construction instead of `where owner=address or sender=address` to help postgres make efficient use of indices
+    WHERE uid = ANY(ARRAY_CAT((SELECT ids FROM regular_orders), (SELECT ids FROM onchain_orders)))
+    
+    UNION ALL
+    
+    -- Get orders from the jit_orders table that match the owner criteria but are not in orders
+    SELECT j.uid, j.kind, j.buy_amount, j.sell_amount, j.fee_amount, j.buy_token, j.sell_token
+    FROM jit_orders j
```
[the remark this answers is not in the document]
Comment (sunce86) [reply crates/orderbook/src/database/total_surplus.rs]: All, but surplus for non-capturing jit orders should be 0 anyway.

File: crates/orderbook/src/database/total_surplus.rs (hunk 1 above)
Comment (sunce86) [reply crates/orderbook/src/database/total_surplus.rs]: Took `AND NOT EXISTS` since the `NOT IN` is slow

**Comment by sunce86** [comment]
> Change looks alright but agree with @squadgazzz's comment.
> 
> > Performance should be good since "jit_orders" table also has index on owners.
> 
> Can you please compare the performance of the surplus query for an account that has many regular orders by manually executing the old and the new query? Would be a shame if we pessimise the performance of the common use case by adding support for jit orders.

I compared it previously with a random owner and it was comparative. But for the sake of mind, I searched for owners with most entries in "orders" table and used them to test the performance.

With owners that have 10-20k entries, this version of code introduces the least performance hit, practically negligible.

**Review by MartinquaXD — APPROVED** [review]

**Review by squadgazzz — APPROVED** [review]

**Pull request merged**